src/index.js

```diff
@@ -1,40 +1,98 @@
+/* eslint-disable no-use-before-define */
+/* eslint-disable max-len */
+/* eslint-disable no-alert */
+/* eslint-disable no-unused-expressions */
 // import _ from 'lodash';
 import './style.css';
 
-const todos = [
-  {
-    index: 1,
-    description: 'Go to the GYM',
-    completed: true,
-  },
-  {
-    index: 2,
-    description: 'Get breakfast',
-    completed: true,
-  },
-  {
-    index: 3,
-    description: 'Start programming',
-    completed: false,
-  },
-  {
-    index: 4,
-    description: 'Go to bed',
-    completed: false,
-  },
-];
-
-todos.sort((a, b) => a.index - b.index);
-
+// select the form element
+const form = document.querySelector('.first-form');
+const todoInput = document.querySelector('.add-task');
 const todoList = document.querySelector('.to-do-list');
 
-for (let i = 0; i < todos.length; i += 1) {
-  todoList.innerHTML += `
+// variables
+let listTodos = JSON.parse(localStorage.getItem('listTodos')) || [];
+
+// FIRST RENDRER
+renderTodo();
+
+// submit form
+form.addEventListener('submit', (e) => {
+  e.preventDefault();
+
+  addTodo();
+  renderTodo();
+  localStorage.setItem('listTodos', JSON.stringify(listTodos));
+  // RESET FORM
+  form.reset();
+});
+
+// add a to do
+function addTodo() {
+  const todoValue = todoInput.value;
+
+  // check if empty to do
+  const isEmpty = todoValue === '';
+
+  // CHEK FOR DUPLICATE TO DO
+  const isDupplicated = listTodos.some((todo) => todo.description.toUpperCase() === todoValue.toUpperCase());
+
+  if (isEmpty) {
+    alert('The to do\'s task is empty');
+  } else if (isDupplicated) {
+    alert('The to do task already exist');
+  } else {
+    const todo = {
+      description: todoValue,
+      completed: false,
+    };
+
+    listTodos.push(todo);
+  }
+}
+
+// RENDER TODO
+function renderTodo() {
+  // CLEAR ELEMENT BEFORE THE RE-RENDER
+  todoList.innerHTML = '';
+
+  // RENDER TODO
+  listTodos.forEach((todo, index) => {
+    todoList.innerHTML += `
   <li>
-    <div class="task-div">
-      <input type="checkbox" />
-      <p>${todos[i].description}</p>
+    <div class="todo" id=${index}>
+      <div class="task-div">
+        <input type="checkbox" />
+        <p>${todo.description}</p>
+      </div>
+      <i class="fa-solid fa-trash" data-action="delete"></i>
     </div>
-    <i class="fa-solid fa-trash"></i>
   </li>`;
+  });
```

Comment:
- [ ] Your todo tasks do not have an `index` property. Kindly add it as stated in the project requirement

![image](https://user-images.githubusercontent.com/98453979/215061397-ee6dbb13-e8e1-493d-a1fe-673bd713966a.png)

### checkout requirement

![image](https://user-images.githubusercontent.com/98453979/215061575-98ec33c3-6888-47dd-81f0-86c392f9813f.png)

- [ ] I also noticed I could not edit your tasks from my end. Kindly ensure your tasks are editable and the changes are saved in the local storage

![image](https://user-images.githubusercontent.com/98453979/215062089-b97223d3-5293-4cc4-871b-e38137a538b2.png)